src/sage/categories/unique_factorization_domains.py

```diff
@@ -167,50 +167,65 @@ def _gcd_univariate_polynomial(self, f, g):
                 sage: g = 4*x + 2
                 sage: f.gcd(g).parent() is R
                 True
+
+            A slightly more exotic base ring::
+
+                sage: P = PolynomialRing(QQbar, 4, "x")
+                sage: p = sum(QQbar.zeta(i + 1) * P.gen(i) for i in range(4))
+                sage: ((p^4 - 1).gcd(p^3 + 1) / (p + 1)).is_unit()
+                True
             """
+            def content(X):
+                """
+                Return the content of ``X`` up to a unit.
+                """
+                X_it = iter(X.coefficients())
```

Comment:
Possible minor optimization:
```suggestion
                X_it = reversed(X.coefficients())
```
seems to be slightly faster, at least for the `Lattes_map` example for the few elliptic curves I tried. Of course in theory it is not more efficient, but maybe monic polynomials are common in practice.

Reply:
Excellent!  Yes, I guess we are more likely to produce monic polynomials if I look at some of the gcd docstrings.